staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -187,6 +209,35 @@ func (a *GenericAdmissionWebhook) loadConfiguration(attr admission.Attributes) (
 	return hookConfig, nil
 }
 
+type versionedAttributes struct {
+	admission.Attributes
+	oldObject runtime.Object
+	object    runtime.Object
+}
+
+func (v versionedAttributes) GetObject() runtime.Object {
+	return v.object
+}
+
+func (v versionedAttributes) GetOldObject() runtime.Object {
+	return v.oldObject
+}
+
+func (a *GenericAdmissionWebhook) convertToGVK(obj runtime.Object, gvk schema.GroupVersionKind) (runtime.Object, error) {
+	if obj.GetObjectKind().GroupVersionKind() == gvk {
```

Comment:
@sttts @deads2k @lavalamp CR fall into this case and thus will skip the conversion.

Reply:
Looks good. On the other hand, I wonder why the convertor does not check the same.

Reply:
please leave a comment to that effect?

Reply:
>  I wonder why the convertor does not check the same.

Actually without this check, it's the `creator` that fails first. It doesn't know how to create a custom resource. \
I read the conversion stack, I don't see any code specially handles the case where `in` and `out` is of the same type. The defaultConvert will generate the correct result though: https://github.com/kubernetes/kubernetes/blob/7650665059e65b4b22375d1e28da5306536a12fb/staging/src/k8s.io/apimachinery/pkg/conversion/converter.go#L577

Reply:
> please leave a comment to that effect?

Added.